File: GithubGists/okhttplibrary/src/main/java/ru/white_monk_team/okhttplibrary/RequestsHandler.java
package ru.white_monk_team.okhttplibrary;

import android.content.Context;

import java.io.IOException;
import java.io.InputStream;
import java.util.Map;
import java.util.Set;

import androidx.annotation.NonNull;
import okhttp3.Request;
import okhttp3.Response;

public class RequestsHandler {

    private Map<String, String> mResponsesMap;
    private Context mContext;

    public RequestsHandler(Context context, Map<String, String> responsesMap) {
        mResponsesMap = responsesMap;
        mContext = context;
    }

    public boolean shouldIntercept(@NonNull String path) {
        Set<String> keys = mResponsesMap.keySet();
        for (String interceptUrl : keys) {
            if (path.contains(interceptUrl)) {
                return true;
            }
        }
        return false;
    }

    @NonNull
    public Response proceed(@NonNull Request request, @NonNull String path) {
        Set<String> keys = mResponsesMap.keySet();
        for (String interceptUrl : keys) {
            if (path.contains(interceptUrl)) {
                String mockResponsePath = mResponsesMap.get(interceptUrl);
                return createResponseFromAssets(request, mockResponsePath);
            }
        }

        return OkHttpResponse.error(request, 500, "Incorrectly intercepted request");
    }

    @NonNull
    private Response createResponseFromAssets(@NonNull Request request, @NonNull String assetPath) {
        try {
            final InputStream stream = mContext.getAssets().open(assetPath);
            try {
                return OkHttpResponse.success(request, stream);
            } finally {
                stream.close();
            }
        } catch (IOException e) {
            return OkHttpResponse.error(request, 500, e.getMessage());
        }
    }
}

File: GithubGists/okhttplibrary/src/main/java/ru/markusfrost/okhttplibrary/RequestsHandler.kt
package ru.markusfrost.okhttplibrary

import android.content.Context
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class RequestsHandler(private val mContext: Context, private val mResponsesMap: Map<String, String>) {

    fun shouldIntercept(path: String): Boolean {
        val keys = mResponsesMap.keys
        for (interceptUrl in keys) {
            if (path.contains(interceptUrl)) {
                return true
            }
        }
        return false
    }

    fun proceed(request: Request, path: String): Response {
        for (interceptUrl in mResponsesMap.keys) {
            if (path.contains(interceptUrl)) {
                return createResponseFromAssets(request,  mResponsesMap[interceptUrl]!!)
            }
        }

        return OkHttpResponse.error(request, 500, "Incorrectly intercepted request")
    }

    private fun createResponseFromAssets(request: Request, assetPath: String): Response {
        try {
            mContext.assets.open(assetPath).use { stream ->
                return OkHttpResponse.success(request, stream)
            }
        } catch (e: IOException) {
            return OkHttpResponse.error(request, 500, e.message!!)
        }

    }
}
